### PR TITLE
[visibility_detector] Adapt to upcoming changes in Flutter

### PR DESCRIPTION
**Pull request by goderbauer**
Prepare the visibility_detector  package for the removal of AbstractNode from the Flutter framework (https://github.com/flutter/flutter/pull/128973).

I had to make a ton of changes unrelated to the above because the stuff in this repo has bit rotted a fair bit and checks were failing otherwise. The changes outside of `packages/visibility_detector/lib/src/render_visibility_detector.dart` are totally unrelated to the above and just general house keeping to make the presubmit checks happy.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I signed the [CLA].
- [x] All tests from running `flutter test` pass.
- [x] `flutter analyze` does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/visibility_detector/pubspec.yaml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 name: visibility_detector
-version: 0.4.0+2
+version: 0.4.0+3
```
Comment (dnfield) [inline packages/visibility_detector/pubspec.yaml:2]: I think this needs to be a bigger bump right?

Comment (goderbauer) [reply packages/visibility_detector/pubspec.yaml:2]: Why's that? All changes are backwards compatible.

**Comment by goderbauer** [comment]
Alright, the presubmit checks made me fix existing issues with `scrollable_positioned_list` and `self_storing_input` as well to get them to pass. These changes are totally unrelated to https://github.com/flutter/flutter/pull/128973. I did them because otherwise the checks are unhappy.

**Review by dnfield — APPROVED** [review]
LGTM

**Pull request merged**